probe-rs/src/bin/probe-rs/util/common_options.rs

```diff
@@ -151,6 +151,9 @@ pub struct ProbeOptions {
     /// firmware, to be erased even when it has read-only protection.
     #[arg(long)]
     pub allow_erase_all: bool,
+    /// Enable vector catch if its supported on the target.
+    #[arg(long)]
+    pub vector_catch: bool,
```

Comment:
I think this shouldn't be part of `ProbeOptions`, if it's only used in the `run` subcommand. It just gets confusing when this is an option everywhere.

And I agree with @Dirbaio  that separate flags for reset and hardfault would be preferable.